plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/flex/CreateAppEngineFlexWtpProject.java

```diff
@@ -46,9 +89,88 @@ public String getDescription() {
+        project, true /* convertIfNecessary */, subMonitor.newChild(50));
+    Set<IFacetedProject.Action> facetInstallSet = new HashSet<>();
+    FacetUtil.addJavaFacetToBatch(JavaFacet.VERSION_1_8, facetedProject, facetInstallSet);
+    FacetUtil.addWebFacetToBatch(WebFacetUtils.WEB_30, facetedProject, facetInstallSet);
+    FacetUtil.addFacetSetToProject(facetedProject, facetInstallSet, subMonitor.newChild(50));
+  }
+
+  private void addDependenciesToProject(IProject project, IProgressMonitor monitor)
       throws CoreException {
-    return CodeTemplates.materializeAppEngineFlexFiles(newProject, config, monitor);
+    SubMonitor subMonitor = SubMonitor.convert(monitor, 100);
+
+    // Create a lib folder
+    IFolder libFolder = project.getFolder("lib");
+    if (!libFolder.exists()) {
+      libFolder.create(true, true, subMonitor.newChild(10));
+    }
+
+    // Download the dependencies from maven
+    M2RepositoryService repoService = new M2RepositoryService();
+    repoService.activate();
+    int ticks = 50 / PROJECT_DEPENDENCIES.size();
+    for (Map.Entry<String, String> dependency : PROJECT_DEPENDENCIES.entrySet()) {
+      LibraryFile libraryFile = new LibraryFile(new MavenCoordinates(dependency.getKey(),
+          dependency.getValue()));
+      Artifact artifact = null;
+      try {
+        artifact = repoService.resolveArtifact(libraryFile, subMonitor.newChild(ticks));
+      } catch (CoreException ex) {
+        logger.log(Level.WARNING, "Error downloading " +
+      libraryFile.getMavenCoordinates().toString() + " from maven", ex);
+        continue;
+      }
+
+      // Copy dependency from local maven repo into lib folder
+      if (artifact != null) {
+        File artifactFile = artifact.getFile();
+        IFile destFile = libFolder.getFile(artifactFile.getName());
+        try {
+          destFile.create(new FileInputStream(artifactFile), true, subMonitor.newChild(30));
+        } catch (FileNotFoundException ex) {
+          logger.log(Level.WARNING, "Error copying over " + artifactFile.toString() + " to " +
+        libFolder.getFullPath().toPortableString(), ex);
+          continue;
+        }
+      }
+    }
+
+    addDependenciesToClasspath(project, libFolder.getLocation().toString(), subMonitor.newChild(10));
+  }
+
+  private void addDependenciesToClasspath(IProject project, String libraryPath,
+      IProgressMonitor monitor)  throws CoreException {
+    IJavaProject javaProject = JavaCore.create(project);
+    IClasspathEntry[] entries = javaProject.getRawClasspath();
+    List<IClasspathEntry> newEntries = new ArrayList<IClasspathEntry>();
+    newEntries.addAll(Arrays.asList(entries));
+
+    // Add all the jars under lib folder to the classpath
+    File libFolder = new File(libraryPath);
+
+    for(File file : libFolder.listFiles()) {
+      IPath path = Path.fromOSString(file.toPath().toString());
+      newEntries.add(JavaCore.newLibraryEntry(path, null, null));
```

Comment:
We should attach the sources and javadocs if available. Probably in a separate PR though.